src/esp/main/io.c

```diff
@@ -10,18 +15,75 @@
 
 #define GPIO_OUTPUT_ENABLE(mask) GPIO.enable_w1ts = (mask)
 
+/**
+The first byte of every request is the protocol identifier.
+The values are assigned defined ranges as follows
 
+0-19 	Reserved
+20-99	Core Protocols
+100-199	Custom Protocols
+200-255	Reserved
+
+*/
+
+#define PROTOCOL_PREFIX_TCPIP	20
+#define PROTOCOL_PREFIX_RS		100
 
 void io_task(void* p)
 {
+  
   while (true) {
-    if (rs_z80_out(read_byte()) != RS_STATE_SEND) {
-      continue;
-    }
-    uint8_t* buf;
-    int size;
-    rs_get_send_buffer(&buf, &size);
-    write_bytes(buf, size);
+	    
+	    uint8_t* buf;
+	    int size;
+	    
+	    uint8_t protocol = read_byte();
+	    
+	    switch( protocol ) {
+		    case PROTOCOL_PREFIX_TCPIP:
+				while (tcp_z80_out(read_byte()) != IP_STATE_SEND_TO_Z80);
+				tcp_get_send_buffer(&buf, &size);
+			    break;
+		    default:
+			    if ( rs_z80_out(protocol) != RS_STATE_SEND )
+			    	while (rs_z80_out(read_byte()) != RS_STATE_SEND);
+				rs_get_send_buffer(&buf, &size);
+			    break;
```

Comment:
Inconsistent indentation.

Reply:
Completely fair comment about tabs/spaces.  I agree.  I paid zero attention to it.  Will pretty it up.